webapp/channels/src/utils/i18n.tsx

```diff
@@ -2,7 +2,25 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {FormattedMessage, type IntlShape, type MessageDescriptor} from 'react-intl';
+import {FormattedMessage, createIntl, createIntlCache, type IntlShape, type MessageDescriptor} from 'react-intl';
+
+import {getCurrentLocale, getTranslations} from 'selectors/i18n';
+import store from 'stores/redux_store';
+
+const cache = createIntlCache();
+
+// getIntl returns an instance of IntlShape for the current locale.
+// Prefer `useIntl` and `FormattedMessage` and only use this selectively when
+// outside of React components.
+export function getIntl(): IntlShape {
```

Comment:
I'm open to the idea, but not entirely sure how to do it apart from copy/pasting the implementation of `IntlProvider` to parameterize and share a common cache and `createIntl` setup. Thoughts?

Reply:
I like that idea. If you want to do that, you could make it so that `IntlProvider` renders an invisible component that just calls `useIntl` and saves the resulting object as a global variable that can be returned by `getIntl`. That way, we should only ever have a single Intl object in the code as long as no one else leaks a reference to it

Reply:
Do we guarantee that the Root component renders before any non-React code needs to access the i18n store? (i.e. error handling or something obscure.) There's got to be an idiomatic way to do this...

Reply:
There's nothing to 100% guarantee it, but the `Root` component is the second component that's rendered after the `App`, so there's very little chance that something that would need the i18n state before that. The error handling code in `entry.tsx` could theoretically use the i18n stuff earlier, but that also expects the Redux store to be initialized, so it may not work before the Root component is mounted already.

I'd be fine making the assumption that this is always initialized in the order we expect, although if we want to be careful, we could add an untranslated warning when `getIntl` would be used before the `IntlProvider` is mounted.

As for doing something idiomatic, `getIntl` itself is already non-idiomatic for us, so I don't think we could make it idiomatic without changing our idioms 😛